File: js/examples/removeCollateral.js
import 'dotenv/config';
import { pathToFileURL } from 'node:url';
import { EZManagerSDK } from '../sdk.js';
import { runReadPosition } from './readPosition.js';

const POSITION_KEY = '';
const COLLATERAL_USDC = '1'; // 1 USDC
const SLIPPAGE_PCT = 0.005; // 0.5%

export async function runRemoveCollateral({
  sdk,
  key = POSITION_KEY,
  usdcAmount = COLLATERAL_USDC,
  slippage = SLIPPAGE_PCT,
  readAfter = true
} = {}) {
  if (!key) throw new Error('POSITION_KEY is required');
  const localSdk = sdk ?? (await EZManagerSDK.fromEnv());

  console.log('Removing collateral...');
  const result = await localSdk.removeCollateral({ key, usdcAmount, slippage });
  console.log(`Collateral removed! Tx: ${result.txHash}`);

  let details = null;
  if (readAfter) {
    details = await runReadPosition({
      sdk: localSdk,
      key,
      label: 'after removeCollateral',
      blockTag: result?.receipt?.blockNumber ?? 'latest'
    });
  }

  return { result, details };
}

const isDirectRun = process.argv[1] && import.meta.url === pathToFileURL(process.argv[1]).href;

if (isDirectRun) {
  await runRemoveCollateral();
}

File: python/examples/removeCollateral.py
from pathlib import Path
import sys

from dotenv import load_dotenv

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))
from sdk import EZManagerSDK
from readPosition import run_read_position

load_dotenv()

POSITION_KEY = ''
COLLATERAL_USDC = '1'  # 1 USDC
SLIPPAGE_PCT = 0.005  # 0.5%


def run_remove_collateral(
    sdk: EZManagerSDK | None = None,
    key: str = POSITION_KEY,
    usdc_amount: str = COLLATERAL_USDC,
    slippage: float = SLIPPAGE_PCT,
    read_after: bool = True,
):
    if not key:
        raise RuntimeError('POSITION_KEY is required')

    local_sdk = sdk or EZManagerSDK.from_env()

    print('Removing collateral...')
    result = local_sdk.remove_collateral(key=key, usdc_amount=usdc_amount, slippage=slippage)
    print(f"Collateral removed! Tx: {result['tx_hash']}")

    details = None
    if read_after:
        post_block = (result.get('receipt') or {}).get('blockNumber', 'latest')
        details = run_read_position(
            sdk=local_sdk,
            key=key,
            label='after removeCollateral',
            block_identifier=post_block,
        )

    return {'result': result, 'details': details}


if __name__ == '__main__':
    run_remove_collateral()